homework-submission/abdullahReact/main.js

```diff
@@ -0,0 +1,176 @@
+const toDoItems = [
+  {
+    id: 1,
+    description: 'Get out of bed',
+    deadLine: '2017-09-11',
+    done: true,
+  },
+  {
+    id: 2,
+    description: 'Brush teeth',
+    deadLine: '2017-09-10',
+    done: false,
+  },
+  {
+    id: 3,
+    description: 'Eat breakfast',
+    deadLine: '2017-09-09',
+    done: false,
+  },
+];
+
+const Label = ({ text, className, htmlFor }) => {
+  return (
+    <label className={className} htmlFor={htmlFor}>
+      {text}
+    </label>
+  );
+};
+
+const Input = ({ name, className, type, placeholder, defaultValue }) => {
+  return (
+    <input
+      name={name}
+      className={className}
+      type={type}
+      placeholder={placeholder}
+      defaultValue={defaultValue}
+    ></input>
+  );
+};
+
+const Header = ({ text, className }) => {
+  return <h2 className={className}>{text}</h2>;
+};
+
+const Button = ({ text, type, className, onClick }) => {
+  return (
+    <button type={type} className={className} onClick={onClick}>
+      {text}
+    </button>
+  );
+};
+
+const Form = ({ addListItem }) => {
+  const x = new Date();
+  const y = x.getFullYear().toString();
+  let m = (x.getMonth() + 1).toString();
+  let d = x.getDate().toString();
+  d.length == 1 && (d = '0' + d);
+  m.length == 1 && (m = '0' + m);
+  let date = y + '-' + m + '-' + d;
```

Comment:
You could move this date generation code to a separate function. So that here you would only write:
```js
let date = getDefaultDate();
```

Reply:
yes it makes sense. I changed my code. now default date is a function